docker/dockerfile-generator.sh

```diff
@@ -0,0 +1,343 @@
+#!/bin/bash
+set -eu
+
+OPENJ9=false
+BUILD=false
+COMMENTS=false
+DOCKERFILE_DIR=
+DOCKERFILE_PATH=
+# Default to JDK8
+JDK_VERSION=8
+JDK_MAX=
+
+setJDKMax() {
+  JDK_MAX=$(wget -q https://api.adoptopenjdk.net/v3/info/available_releases -O - \
+	  | grep 'most_recent_feature_version' \
+	  | cut -d':' -f 2 \
+	  | sed 's/,//g; s/ //g')
+}
+
+processArgs() {
+  local arg
+  local cleanRepo
+  while [[ $# -gt 0 ]]
+  do	
+    arg="$1"
+    case $arg in
+      -h | --help)
+        usage
+        exit 0
+        ;;
+      --openj9)
+        OPENJ9=true
+        shift
+        ;;
+      --build)
+        BUILD=true
+        shift
+        ;;
+      --clean)
+        cleanRepo=true
+        shift
+        ;;
+      --comments)
+        COMMENTS=true
+        shift
+        ;;
+      --path)
+        DOCKERFILE_DIR=$2
+        shift
+        shift
+        ;;
+      --print)
+        PRINT=true
+        shift
+        ;;
+      --jdk)
+        JDK_VERSION="$2"
+        checkJDK
+        shift
+        shift
+        ;;
+      *)
+        echo "Unrecognised Argument: $1"
+        exit 1
+        ;;
+    esac
+  done
+ 
+  if [ -z "$DOCKERFILE_DIR" ]; then
+    DOCKERFILE_DIR=$PWD
+  fi
+
+  if [ ${cleanRepo} ]; then
+    echo "Removing Dockerfile* from $DOCKERFILE_DIR" && rm -rf $DOCKERFILE_DIR/Dockerfile*
+  fi
+
+  DOCKERFILE_PATH="$DOCKERFILE_DIR/Dockerfile"
+  if [ ${OPENJ9} == true ]; then
+    DOCKERFILE_PATH="$DOCKERFILE_PATH-openj9"
+  fi
+}
+
+usage() {
+  echo" Usage: ./dockerfile_generator.sh [OPTIONS]
+  Options:
+	  --help | -h		Print this message and exit
+	  --build		Build the docker image after generation and create interactive container
+	  --clean		Remove all dockerfiles (Dockerfile.JDK*) from '--path'
+	  --comments		Prints comments into the dockerfile
+	  --jdk			Specify which JDK the docker image will be able to build (Default: 8)
+	  --path <FILEPATH>	Specify where to save the dockerfile (Default: $PWD)
+	  --print		Print the Dockerfile to screen after generation
+	  --openj9		Make the Dockerfile able to build w/OpenJ9 JIT"
+}
+
+# Checks to ensure the input JDK is valid
+checkJDK() {
+  if [ ${JDK_VERSION} == "next" ]; then
+    JDK_VERSION=${JDK_MAX}
```

Comment:
Is next what we really get back?

Reply:
With latest commit, this has been changed to just `jdk`, as that's in keeping with `makejdk-any-platform.sh`. For the sake of the rest of the script, this is referred to the number returned by the api under `most_recent_feature_version` (from the link Stewart posted above)